projects/plugins/jetpack/extensions/blocks/ai-assistant/get-suggestion-with-stream.js

```diff
@@ -89,31 +93,38 @@ export async function requestToken() {
 
 	const apiNonce = window.JP_CONNECTION_INITIAL_STATE.apiNonce;
 	const siteSuffix = window.JP_CONNECTION_INITIAL_STATE.siteSuffix;
-	const isJetpackSite = ! window.wpcomFetch;
+	const hasWPComFetch = ! window.wpcomFetch;
 	let data;
 
-	if ( isJetpackSite ) {
-		data = await apiFetch( {
-			path: '/jetpack/v4/jetpack-ai-jwt?_cacheBuster=' + Date.now(),
-			credentials: 'same-origin',
-			headers: {
-				'X-WP-Nonce': apiNonce,
-			},
-			method: 'POST',
-		} );
-	} else {
-		data = await apiFetch( {
-			path: '/wpcom/v2/sites/' + siteSuffix + '/jetpack-openai-query/jwt',
-			method: 'POST',
-		} );
+	try {
+		if ( ! isSimpleSite() && ! isAtomicSite() ) {
+			// Jetpack sites
+			data = await apiFetch( {
+				path: '/jetpack/v4/jetpack-ai-jwt?_cacheBuster=' + Date.now(),
+				credentials: 'same-origin',
+				headers: {
+					'X-WP-Nonce': apiNonce,
+				},
+				method: 'POST',
+			} );
+		} else {
+			// Simple and Atomic sites
+			data = await apiFetch( {
+				path: '/wpcom/v2/sites/' + siteSuffix + '/jetpack-openai-query/jwt',
+				method: 'POST',
+			} );
+		}
+	} catch ( err ) {
```

Comment:
I don't see how this catch is adding any value. It just catches and throws the same error without any other action.
The result is that the code in `askJetpack` (which is calling `akQuestion`, and then `getToken`) will still catch the same error generated by apiFetch failing

```
	try {
		source = await askQuestion( question );
	} catch ( err ) {
		debug( 'Error', err );
		return source;
	}
```

Reply:
Right. Thanks for the feedback. Updated.